python/cuda_parallel/cuda/parallel/experimental/__init__.py

```diff
@@ -194,14 +198,15 @@ def _dtype_validation(dt1, dt2):
 
 
 class _Reduce:
-    def __init__(self, d_in, d_out, op, init):
+    # TODO: constructor shouldn't require concrete `d_in`, `d_out`:
```

Comment:
That's what I was wondering about, but I didn't get to drilling down.

It might be useful to work together on getting this TODO done.

Already, this PR will have complicated merge conflicts with my #2788, i.e. it might be best to team up working on both.

Reply:
> It might be useful to work together on getting this TODO done.

Definitely! I opened https://github.com/NVIDIA/cccl/issues/3008 to track this. Let's tackle it as a follow up to this PR.